app/core/settings.py

```diff
@@ -88,6 +92,55 @@
     'DATETIME_FORMAT': "%Y-%m-%dT%H:%M:%SZ",
 }
 
+LOG_ATTRS = ['asctime', 'created', 'levelname', 'message',
+             'filename', 'funcName', 'lineno', 'module', 'pathname']
+LOG_FORMAT_STR = ' '.join(['%(' + v + ')s' for v in LOG_ATTRS])
+
+LOGGING = {
+    'version': 1,
+    'disable_existing_loggers': False,
+    'formatters': {
+        'json': {
+            '()': jsonlogger.JsonFormatter,
+            'format': LOG_FORMAT_STR,
+        },
+        'verbose': {
+            'format': "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] %(message)s",
+            'datefmt': "%Y-%b-%dT%H:%M:%SZ"
+        },
+        'simple': {
+            'format': '%(levelname)s %(message)s'
+        },
+    },
+    'handlers': {
+        'file': {
+            'level': 'DEBUG',
+            'class': 'logging.FileHandler',
+            'filename': os.path.join(DEFAULT_LOG_DIR, '%s.json.log' % PROJECT_NAME),
+            'formatter': 'json'
```

Comment:
sample taken locally:
```
{
  "asctime": "2018-08-06 08:54:59,978",
  "created": 1533545699.9784408,
  "levelname": "DEBUG",
  "message": "setting request as user cannot view unpublished content/cannot preview",
  "filename": "middleware.py",
  "funcName": "middleware",
  "lineno": 36,
  "module": "middleware",
  "pathname": "./core/middleware.py"
}
```
- `asctime`: `2018-08-06T08:54:59.987` would be more standard, milliseconds are an addition to the standard but they are useful in the debugging context
- `created` can probably be removed` if it's not an hassle to do it`
- `levelname`, `message` good
- `funcName`, `lineno`, `module`, `pathname` in concert lead me to understand where the log line was produced from so they work

Reply:
Sample from a `.exception()` logging statement adds this:
```
  "exc_info": "Traceback (most recent call last):\n  File \"/srv/digests/venv/lib/python3.6/site-packages/django/core/handlers/exception.py\", line 34, in inner\n    response = get_response(request)\n  File \"/srv/digests/venv/lib/python3.6/site-packages/django/core/handlers/base.py\", line 126, in _get_response\n    response = self.process_exception_by_middleware(e, request)\n  File \"/srv/digests/venv/lib/python3.6/site-packages/django/core/handlers/base.py\", line 124, in _get_response\n    response = wrapped_callback(request, *callback_args, **callback_kwargs)\n  File \"/srv/digests/venv/lib/python3.6/site-packages/django/views/decorators/csrf.py\", line 54, in wrapped_view\n    return view_func(*args, **kwargs)\n  File \"/srv/digests/venv/lib/python3.6/site-packages/rest_framework/viewsets.py\", line 103, in view\n    return self.dispatch(request, *args, **kwargs)\n  File \"/srv/digests/venv/lib/python3.6/site-packages/rest_framework/views.py\", line 483, in dispatch\n    response = self.handle_exception(exc)\n  File \"/srv/digests/venv/lib/python3.6/site-packages/rest_framework/views.py\", line 443, in handle_exception\n    self.raise_uncaught_exception(exc)\n  File \"/srv/digests/venv/lib/python3.6/site-packages/rest_framework/views.py\", line 480, in dispatch\n    response = handler(request, *args, **kwargs)\n  File \"./digests/api.py\", line 56, in create\n    self._validate_against_schema(request, data=request.data)\n  File \"./digests/api.py\", line 42, in _validate_against_schema\n    schema = get_schema(get_schema_name(request.content_type))\n  File \"./digests/utils.py\", line 11, in get_schema\n    with open(os.path.join(SCHEMA_DIRECTORY, schema_name)) as schema:\nFileNotFoundError: [Errno 2] No such file or directory: /srv/digests/venv/lib/python3.6/site-packages/elife_api_validator/schemas/.v1.json",
  "status_code": 500,
  "request": "<WSGIRequest: POST /digests>"
```
`exc_info` is standard, also the others are understandable so if this is a Django format ok.